selene-lib/src/lints/roblox_suspicious_udim2_new.rs

```diff
@@ -0,0 +1,138 @@
+use super::*;
+use crate::ast_util::range;
+use std::convert::Infallible;
+
+use full_moon::{
+    ast::{self, Ast},
+    visitors::Visitor,
+};
+
+pub struct UDim2ArgCountLint;
+
+impl Lint for UDim2ArgCountLint {
+    type Config = ();
+    type Error = Infallible;
+
+    const SEVERITY: Severity = Severity::Warning;
+    const LINT_TYPE: LintType = LintType::Correctness;
+
+    fn new(_: Self::Config) -> Result<Self, Self::Error> {
+        Ok(UDim2ArgCountLint)
+    }
+
+    fn pass(&self, ast: &Ast, context: &Context, _: &AstContext) -> Vec<Diagnostic> {
+        if !context.is_roblox() {
+            return Vec::new();
+        }
+
+        let mut visitor = UDim2CountVisitor::default();
+
+        visitor.visit_ast(ast);
+
+        visitor
+            .args
+            .iter()
+            .map(|mismatch| {
+                Diagnostic::new_complete(
+                    "roblox_suspicious_udim2_new",
+                    format!(
+                        "UDim2.new takes 4 numbers, but {} {} provided.",
+                        mismatch.args_provided,
+                        if mismatch.args_provided == 1 {
+                            "was"
+                        } else {
+                            "were"
+                        }
+                    ),
+                    Label::new(mismatch.call_range),
+                    vec![
+                        if mismatch.args_provided > 2 || !mismatch.args_are_numbers {
+                            ""
```

Comment:
We shouldn't pass anything in this vector if this branch is hit, it's just a bit weird.

The cleanest way to do this would probably be turning this whole thing into a function that you can more easily return from